soc/espressif/esp32s3/esp32s3-mp.c

```diff
@@ -35,7 +38,135 @@ void esp_appcpu_start(void *entry_point)
 
 	esp_rom_ets_set_appcpu_boot_addr((void *)entry_point);
 
-	ets_delay_us(50000);
+	esp_cpu_reset(1);
+}
+
+static int load_segment(uint32_t src_addr, uint32_t src_len, uint32_t dst_addr)
+{
+	const uint32_t *data = (const uint32_t *)sys_mmap(src_addr, src_len);
+
+	if (!data) {
+		ets_printf("%s: mmap failed", __func__);
+		return -1;
+	}
+
+	volatile uint32_t *dst = (volatile uint32_t *)dst_addr;
+
+	for (int i = 0; i < src_len / 4; i++) {
+		dst[i] = data[i];
+	}
```

Comment:
Just to avoid problems with cache coherence, it would not be good to invalidate D-Cache here before each time we move the data? Of course just in case we're moving data from the flash to the DRAM.

Reply:
Since updating ld script for the APPCPU did not get the `STORE_PROHIBITED` exception anymore, and I'd rather not change this now.